### ISSUE TITLE
Java packages hanging on install

### ISSUE THREAD

**Issue by noamross** [cardcorp/card-rocker#1]
I am unsure whether this is an issue with your container, but I'm having a problem with both your container and rocker/hadleyverse (which uses an older version of Java).  When, in the container, I install a package which has an rJava dependency, the install hangs on 

```
testing if installed package can be loaded
```

I think this is occurring when `.jpackage()` is called in the package `.onLoad`.

Similarly, when I run `R CMD javareconf`, even with the `-n` flag, in the terminal in the container, the command hangs after reading out

```
`Java interpreter : /usr/bin/java`
```

I've tried both of the above both with and without first doing this:

```
export JAVA_HOME=/usr/lib/jvm/VARIOUS_JAVA_INSTALLS
export PATH=$PATH:$JAVA_HOME/bin
```

@cboettig @daroczig

**Comment by daroczig** [comment]
Not sure what the problem might be -- I've just pulled the most recent image and tried `R CMD javareconf` to reproduce this (although we are using quite a few R packages with Java dependencies on a daily basis without any issue), but got this working:

``` r
root@gergely-CARD:/var/lib# docker pull cardcorp/r-java
Using default tag: latest
latest: Pulling from cardcorp/r-java
4bdbb6a8cd61: Pull complete 
d055e9fc6f5e: Pull complete 
81b0d56d86ad: Pull complete 
788a362960f8: Pull complete 
47ba62e1d6ac: Pull complete 
2f536fe77541: Pull complete 
934a40a43a73: Pull complete 
d7ee5ec02faf: Pull complete 
7124b5abc91a: Pull complete 
c953e8f2aa41: Pull complete 
c65e5ae46826: Pull complete 
9e6a996f8304: Pull complete 
0731f799538a: Pull complete 
bb58f4406ed6: Pull complete 
180f3a769ebf: Pull complete 
Digest: sha256:a4a8088a157c318818769cfe520ff3db3a28b0845fd87f676161482d37481cd0
Status: Downloaded newer image for cardcorp/r-java:latest

root@gergely-CARD:/var/lib# docker run --rm -ti cardcorp/r-java bash

root@29f572311706:/# R CMD javareconf
*** JAVA_HOME is not a valid path, ignoring
Java interpreter : /usr/bin/java
Java version     : 1.8.0_66
Java home path   : /usr/lib/jvm/java-8-oracle/jre
Java compiler    : /usr/bin/javac
Java headers gen.: /usr/bin/javah
Java archive tool: /usr/bin/jar

trying to compile and link a JNI program 
detected JNI cpp flags    : -I$(JAVA_HOME)/../include -I$(JAVA_HOME)/../include/linux
detected JNI linker flags : -L$(JAVA_HOME)/lib/amd64/server -ljvm
gcc -std=gnu99 -I/usr/share/R/include -DNDEBUG -I/usr/lib/jvm/java-8-oracle/jre/../include -I/usr/lib/jvm/java-8-oracle/jre/../include/linux     -fpic  -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c conftest.c -o conftest.o
gcc -std=gnu99 -shared -L/usr/lib/R/lib -Wl,-z,relro -o conftest.so conftest.o -L/usr/lib/jvm/java-8-oracle/jre/lib/amd64/server -ljvm -L/usr/lib/R/lib -lR


JAVA_HOME        : /usr/lib/jvm/java-8-oracle/jre
Java library path: $(JAVA_HOME)/lib/amd64/server
JNI cpp flags    : -I$(JAVA_HOME)/../include -I$(JAVA_HOME)/../include/linux
JNI linker flags : -L$(JAVA_HOME)/lib/amd64/server -ljvm
Updating Java configuration in /usr/lib/R
Done.
```

Maybe it's the host OS or `docker` version? I've tested on Ubuntu 12.04 & 15.10

**Comment by noamross** [comment]
My host OS was boot2docker on OSX.  I found that the problem was a known issue with java on boot2docker 1.9.1, which should be fixed on the next version but is temporarily fixed by downgrading to 1.9.0.  Revelant issue here in case you are interested: https://github.com/boot2docker/boot2docker/issues/1099

Thanks for maintaining this container!

**Comment by daroczig** [comment]
Thanks a lot, @noamross, for the update, much appreciated and good to know what the issue was.
